client/src/test/java/com/sinch/sdk/domains/voice/adapters/ApplicationsServiceTest.java

```diff
@@ -0,0 +1,126 @@
+package com.sinch.sdk.domains.voice.adapters;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.adelean.inject.resources.junit.jupiter.TestWithResources;
+import com.sinch.sdk.BaseTest;
+import com.sinch.sdk.core.exceptions.ApiException;
+import com.sinch.sdk.core.http.AuthManager;
+import com.sinch.sdk.core.http.HttpClient;
+import com.sinch.sdk.domains.voice.adapters.api.v1.ApplicationsApi;
+import com.sinch.sdk.domains.voice.adapters.converters.ApplicationsDtoConverter;
+import com.sinch.sdk.domains.voice.adapters.converters.ApplicationsDtoConverterTest;
+import com.sinch.sdk.domains.voice.models.CallbackUrls;
+import com.sinch.sdk.domains.voice.models.dto.v1.ApplicationsCallbackUrlsDtoTest;
+import com.sinch.sdk.domains.voice.models.dto.v1.ApplicationsGetNumbersResponseDtoTest;
+import com.sinch.sdk.domains.voice.models.dto.v1.CallbacksDto;
+import com.sinch.sdk.domains.voice.models.dto.v1.UnassignNumbersDto;
+import com.sinch.sdk.domains.voice.models.dto.v1.UpdateNumbersDto;
+import com.sinch.sdk.domains.voice.models.response.AssignedNumbers;
+import com.sinch.sdk.models.Configuration;
+import com.sinch.sdk.models.E164PhoneNumber;
+import java.util.Map;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+
+@TestWithResources
+public class ApplicationsServiceTest extends BaseTest {
+
+  @Mock ApplicationsApi api;
+  @Mock Configuration configuration;
+  @Mock HttpClient httpClient;
+  @Mock Map<String, AuthManager> authManagers;
+
+  @Captor ArgumentCaptor<String> applicationKeyCaptor;
+  @Captor ArgumentCaptor<CallbacksDto> callbacksDtoCaptor;
+  @Captor ArgumentCaptor<UpdateNumbersDto> updateNumbersDtoCaptor;
+  @Captor ArgumentCaptor<UnassignNumbersDto> unassignNumbersDtoCaptor;
+
+  ApplicationsService service;
+
+  @BeforeEach
+  public void initMocks() {
+    service = spy(new ApplicationsService(configuration, httpClient, authManagers));
+    doReturn(api).when(service).getApi();
+  }
+
+  @Test
+  void getNumbers() throws ApiException {
+
+    when(api.configurationGetNumbers()).thenReturn(ApplicationsGetNumbersResponseDtoTest.expected);
```

Comment:
Just to advertise [mock server](https://github.com/sinch/doppelganger/blob/f6c70b6a61a51e06eba1a0b70cb878d57f8e427b/voice/applications.js#L6) - it's possible to reuse that mock for the test. 😉

Reply:
@Dovchik in the mockserver, the mock files as I see them as subobject of objects hardcoded in javascript files. Are you saying it exists a json version of them?

Reply:
@asein-sinch it's more of a nudge to use mockserver and point request to it
Answering your question: no, but it's possible to have them as a json files if we need it.